### ISSUE TITLE
Starting VSCode at Linux Ubuntu are too much slower

### ISSUE THREAD

**Issue by ldeitos** [Microsoft/vscode#15644]
- VSCode Version: 1.7.1
- OS Version: #88-Ubuntu SMP Tue Dec 3 18:00:02 UTC 2013

Steps to Reproduce:

1. Start up VSCode spend up to 24' (minutes)

![startup_console_1](https://cloud.githubusercontent.com/assets/8965515/20390815/a60f0b84-acb7-11e6-9393-717387a004ed.png)
![startup_console_2](https://cloud.githubusercontent.com/assets/8965515/20390814/a607a7ea-acb7-11e6-8c7b-c5f5bcef69b8.png)

**Comment by bpasero** [comment]
@ldeitos thanks so much for trying to run with --performance. It looks like all the time is spend on the main side of Code way before we even try to open a window. I wonder where that time is spend and unfortunately we do not have more detailed info in that area. 

Do you see any output when you run `code --verbose` from the commandline?

**Comment by ldeitos** [comment]
@bpasero thanks to quick return! Unfortunately doesn't have any return at console stating with --verbose option.

**Comment by bpasero** [comment]
Please see if you can find any logged output that could help us find the source of the issue:
* run `code --verbose` from the command line and attach the output
* in Code select **Help | Toggle Developer Tools** and select the **Console**
* attach any output you see in the **Console** you can find

Thanks!

**Comment by ldeitos** [comment]
Hi @bpasero, 

After a long wait...

OS console output:
```shell
[main 14:07:10] Starting VS Code in verbose mode
[main 14:07:10] from: /home/02844823904/Instaladores/VSCode-linux-ia32/resources/app
[main 14:07:10] args: { _: [],
  help: false,
  h: false,
  version: false,
  v: false,
  wait: false,
  w: false,
  diff: false,
  d: false,
  goto: false,
  g: false,
  'new-window': false,
  n: false,
  'reuse-window': false,
  r: false,
  performance: false,
  p: false,
  verbose: true,
  logExtensionHostCommunication: false,
  'disable-extensions': false,
  disableExtensions: false,
  'list-extensions': false,
  'show-versions': false,
  nolazy: false }
[main 14:07:13] IPC#vscode:startCrashReporter
[main 14:07:13] IPC#vscode:setHeaders
[main 14:07:14] IPC#vscode-workbenchLoaded
```

Apparently logs generated by `--verbose` option only starts to print a long time after send `code` start command.

VSCode console output: 

```
Error: /usr/lib/i386-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.18' not found (required by /home/02844823904/Instaladores/VSCode-linux-ia32/resources/app/node_modules/native-keymap/build/Release/keymapping.node)(…)
Failed to clear temp storage: It was determined that certain files are unsafe for access within a Web application, or that too many calls are being made on file resources. SecurityError
```

**Comment by bpasero** [comment]
@ldeitos is that output here:

`[main 14:07:10] Starting VS Code in verbose mode`

coming after 20 minutes only?

**Comment by ldeitos** [comment]
Yep

**Comment by bpasero** [comment]
W.O.W

**Comment by bpasero** [comment]
@ldeitos how about running just the Electron framework? this is the tool we use for all UI. You could get a version to double click from here: https://github.com/electron/electron/releases/tag/v1.3.7

Would be interesting to know if that opens also only after 20 minutes.

**Comment by ldeitos** [comment]
Ok, I will try, but only tomorow... Thanks!

**Comment by ldeitos** [comment]
Hi, I just download and run the Electron 1.3.7 standalone and that open instantaneously...
After that, I call `./electron ~/Instaladores/VSCode-linux-ia32/resources/app` command to run VSCode and the result was the same: take 20+ min to open.
I try open another's simple applications through Electron and thats are fast to open... Any idea?

**Comment by bpasero** [comment]
@joaomoreno any clue what could take 20 minutes until it reaches our `main.ts#main()` function? 

I was first suspecting this to be caused by our CLI scripts, but it seems it also happens when you just run electron. 


/cc @Tyriar for any clues?

**Comment by Tyriar** [comment]
@ldeitos so this is regular 32-bit Ubuntu?

This looks concerning:

```
Error: /usr/lib/i386-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.18' not found (required by /home/02844823904/Instaladores/VSCode-linux-ia32/resources/app/node_modules/native-keymap/build/Release/keymapping.node)(…)
```

We do require GLIBCXX 3.4.15+ https://code.visualstudio.com/Docs/supporting/requirements#_additional-linux-requirements

**Comment by ldeitos** [comment]
@Tyriar, 
Unfortunately not... the Ubuntu is customized by operation area from company where I work... maybe that's the  problem.

**Comment by joel-wright** [comment]
I believe I may be seeing something related to this issue on Ubuntu 16.10.

I have never managed to wait long enough for `/usr/share/code/code` to start - it just appears to hang and do nothing, but my process list shows that a number of vscode related processes have kicked off:

```jjw      20251  4449  6 23:52 pts/9    00:00:00 /usr/share/code/code --performance --verbose
jjw      20253 20251  0 23:52 pts/9    00:00:00 /usr/share/code/code --type=zygote --no-sandbox
jjw      20276 20251  0 23:52 pts/9    00:00:00 /usr/share/code/code --type=gpu-process --channel=20251.0.528142247 --mojo-application-channel-token=1AD72241F21E26AA8408450C08A0CCE2 --no-sandbox --window-depth=24 --x11-visual-id=33 --supports-dual-gpus=false --gpu-driver-bu rounds=5,18,56,70 --gpu-vendor-id=0x10de --gpu-device-id=0x1c20 --gpu-driver-vendor --gpu-driver-version --gpu-driver-date --gpu-secondary-vendor-ids=0x8086 --gpu-secondary-device-ids=0x191b --v8-natives-passed-by-fd --v8-snapshot-passed-by-fd
```

The weird thing is, if I start vscode using `/usr/bin/code` everything starts immediately and normally. I only discovered this by kicking off vscode from the command line rather than using the .desktop added to unity.

**Comment by Tyriar** [comment]
Hmm, I still haven't been able to repro this on 16.10. The difference between the 2 commands are:

- via terminal: `ELECTRON_RUN_AS_NODE=1 /usr/share/code/code /usr/share/code/resources/app/out/cli.js `
- via .desktop `/usr/share/code/code`

@bpasero @joaomoreno do you expect there to be a slowdown when just calling the electron binary directly without `ELECTRON_RUN_AS_NODE=1`?

**Comment by joel-wright** [comment]
I'm happy to help with tracking down this issue if I can, but running `/usr/share/code/code --verbose --performance` just produces the following 2 lines of output and hangs for >30mins:

```
bash: cannot set terminal process group (-1): Inappropriate ioctl for device
bash: no job control in this shell
```

Let me know if there's anything else I can do...

**Comment by joel-wright** [comment]
Setting the env var `ELECTRON_RUN_AS_NODE=1` and running `/usr/share/code/code` takes me immediately to a prompt.

**Comment by ldeitos** [comment]
I will try as soon as possible... I'm at vacation just now... on after 15/01 I will return and test the solution.

**Comment by joel-wright** [comment]
My startup issues seem to have been resolved with the 1.9.x release.

**Comment by ldeitos** [comment]
@joel-wright, sorry for the delay in responding!

Set env var as you recommended worked well, thanks for your attention.

As well, really after version 1.9 bug was fixed without set the variable.
